factoryWithTypeCheckers.js

```diff
@@ -434,6 +441,9 @@ module.exports = function(isValidElement, throwOnDirectAccess) {
       var allKeys = assign({}, props[propName], shapeTypes);
       for (var key in allKeys) {
         var checker = shapeTypes[key];
+        if (shapeTypes.hasOwnProperty(key) && typeof checker !== 'function') {
```

Comment:
Please never rely on `hasOwnProperty` being on an object directly; use `has` instead
```suggestion
        if (has(shapeTypes, key) && typeof checker !== 'function') {
```

Reply:
👍